core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -853,6 +857,18 @@ public void install() {
                     LOGGER.log(Level.WARNING, "Dependent install of " + dep.name + " for plugin " + name + " already added, skipping");
                 }
             }
+            PluginWrapper pw = getInstalled();
+            if(pw != null) { // JENKINS-34494 - check for this plugin being disabled
+                Future<UpdateCenterJob> enableJob = null;
+                if(!pw.isEnabled()) {
```

Comment:
If you, say from the setup wizard, call `PluginManager.installPlugins`, it will blindly start calling `deploy` on them, disabled plugins will still remain disabled, this handles the case for both dependencies and directly installed plugins.